src/arrayMethodSort.js

```diff
@@ -4,8 +4,43 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompare = (a, b) => {
+      const strA = String(a);
+      const strB = String(b);
+
+      if (strA < strB) {
+        return -1;
+      }
+
+      if (strA > strB) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    const compare = compareFunction || defaultCompare;
+
+    const array = this.slice();
```

Comment:
Using `this.slice()` is a good way to create a shallow copy of the array, but be aware that it doesn't copy nested arrays or objects. If the array contains nested structures, you might need a deep copy depending on the requirements of the sort function.